torch/csrc/jit/interpreter.cpp

```diff
@@ -193,6 +362,55 @@ struct CodeImpl {
   std::vector<bool> bool_data;
 };
 
+
+struct Registers {
+  Registers(size_t size)
+  : registers(size) {}
+  Registers(const Registers & rhs)
+  : registers(rhs.registers) {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->retain();
+    }
+  }
+  ~Registers() {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->release();
+    }
+  }
+  at::RefCounted*& operator[](size_t i) {
+    return registers[i];
+  }
+  //guarenteed to be non-null
```

Comment:
What is guaranteed to be non-null? You have a `isValid` check inside, so it seems like it could be null.

Reply:
(no longer exists)